lib/util/template/index.ts

```diff
@@ -162,28 +161,35 @@ const allowedFieldsList = Object.keys(allowedFields)
 
 type CompileInput = Record<string, unknown>;
 
-type FilteredObject = Record<string, CompileInput | CompileInput[] | unknown>;
-
-function getFilteredObject(input: CompileInput): FilteredObject {
-  const obj = clone(input);
-  const res: FilteredObject = {};
-  const allAllowed = [
-    ...Object.keys(allowedFields),
-    ...exposedConfigOptions,
-  ].sort();
-  for (const field of allAllowed) {
-    const value = obj[field];
+const allowedTemplateFields = new Set([
+  ...Object.keys(allowedFields),
+  ...exposedConfigOptions,
+]);
+
+const compileInputProxyHandler: ProxyHandler<CompileInput> = {
+  get(target: CompileInput, prop: keyof CompileInput): unknown {
+    if (!allowedTemplateFields.has(prop)) {
+      return undefined;
```

Comment:
@rarkins  should we warn here?, so users know  they have errors in templates?

with this we can remove the manual template check with regex.

Reply:
If so, let's do this in separate PR maybe?